src/service/src/local.rs

```diff
@@ -36,23 +37,52 @@ impl Activatable for Thread {
     }
 }
 
+/// An activator for a thread.
+///
+/// This wraps any `Activatable` and has a `Drop` impl to ensure the thread is always activated
+/// when the activator is dropped. This is important to ensure workers have a chance to observe
+/// that their command channel has closed and prepare for reconnection.
+#[derive(Debug)]
+pub struct LocalActivator {
+    inner: Box<dyn Activatable>,
+}
```

Comment:
I avoided making this type generic over `Activatable` to reduce some of the noise in the using code.